### PR TITLE
feat: Search in scrap menu

### PR DESCRIPTION
**Pull request by ryanzhoudev**

### CODE REVIEW

**Comment by magicus** [comment]
How is the layout of the scrap menu? Do we need to adjust the "searchable content area" so we don't find items that are part of the UI?

**Comment by ryanzhoudev** [comment]
Yes, the top bar (items 0-8) are ui.

**Comment by magicus** [comment]
I think we should try to restore the addition of a "content area" definition in SearchableContainerType, which was there for a while during the PR review that introduced the fix in matchItems, instead of hard-coding yet another exception. Do you feel up for it, or should I create the abstraction so you just can add the scrap menu?

**Comment by ryanzhoudev** [comment]
I would like to give it a try

File: common/src/main/java/com/wynntils/models/containers/type/ContainerBounds.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    private final List<Integer> slots = new ArrayList<>();
+
+    /**
+     * @param totalRows Total number of rows in the container
+     * @param startRow 0-indexed row of the first row containing items
+     * @param startCol 0-indexed column of the first column containing items
+     * @param endRow 0-indexed row of the last row containing items
+     * @param endCol 0-indexed column of the last column containing items
+     */
+    public ContainerBounds(int totalRows, int startRow, int startCol, int endRow, int endCol) {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/models/containers/type/ContainerBounds.java]: Do you really need totalRows? Afaict it is only used to determine when to stop looping over slots, but for that you can use endRow, right?

Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/models/containers/type/ContainerBounds.java]: Oh yeah

**Review by magicus — APPROVED** [review]

**Pull request merged**